packages/fx-core/src/common/local/projectSettingsHelper.ts

```diff
@@ -14,20 +14,34 @@ import {
 import { isAADEnabled, IsSimpleAuthEnabled } from "../tools";
 import { ResourcePlugins } from "../constants";
 import { BotCapabilities, BotHostTypeName, BotHostTypes } from "./constants";
+import { isV3 } from "../../core";
+import {
+  hasAAD,
+  hasBot,
+  hasFunction,
+  hasFunctionBot,
+  hasSimpleAuth,
+  hasTab,
+  isSPFx,
+} from "../projectSettingsHelperV3";
 
 export class ProjectSettingsHelper {
   // keep the same logic as plugin.activate()
   public static isSpfx = (projectSettings: ProjectSettings | undefined): boolean =>
-    (projectSettings?.solutionSettings as AzureSolutionSettings)?.hostType ===
-    HostTypeOptionSPFx.id;
+    isV3()
+      ? isSPFx(projectSettings as ProjectSettingsV3)
+      : (projectSettings?.solutionSettings as AzureSolutionSettings)?.hostType ===
+        HostTypeOptionSPFx.id;
 
   public static includeFrontend(projectSettings: ProjectSettings | undefined): boolean {
+    if (isV3()) return hasTab(projectSettings as ProjectSettingsV3);
```

Comment:
If the project is SPFx, what will the result of `hasTab` be?

Reply:
hasTab is true for SPFx Tab

Reply:
Currently, `includeFrontend` means including Azure Tab, not SPFx Tab.

Reply:
OK, I will replace it with hasAzureTab()